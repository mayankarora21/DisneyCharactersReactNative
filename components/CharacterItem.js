import {View, Text, Image, StyleSheet, Pressable, ScrollView} from 'react-native';
import {useNavigation} from '@react-navigation/native';

const CharacterItem = ({character}) => {
    const navigation = useNavigation();
    
    const handlePress = () => {
        navigation.navigate('Details', {
            character: character
        });
    }

    return (
        <Pressable style = {styles.characterContainer} onPress = {handlePress}>
            <Text>{character.name}</Text>
            <Image source={{uri: character.imageUrl}} style = {styles.image}></Image>
        </Pressable>
    );
}

export default CharacterItem;


const styles = StyleSheet.create({
    characterContainer: {
        borderBottomWidth: 1,
        marginBottom: 10,
        padding: 5,
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    image: {
        height: 100,
        width: 100,
        resizeMode: 'stretch',
        borderRadius: 50
    }
})