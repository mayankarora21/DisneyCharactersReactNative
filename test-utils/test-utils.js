import React from 'react'
import {render} from '@testing-library/react-native';
import store from '../store/store';
import { Provider } from 'react-redux';
import {NavigationContainer} from '@react-navigation/native';

const AllTheProviders = ({children}) => {
  return (
    <Provider store = {store}>
      <NavigationContainer>
            {children}
      </NavigationContainer>
    </Provider>
  )
}

const customRender = (ui, options) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}