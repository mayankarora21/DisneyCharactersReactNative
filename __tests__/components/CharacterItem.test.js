import CharacterItem from '../../components/CharacterItem';
import {render, screen} from '../../test-utils/test-utils';

test('CharacterItem should render correctly', () => {
    render(<CharacterItem character = {{name: 'CharacterName', imageUrl: 'https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png'}}/>)
    expect(screen.toJSON()).toMatchSnapshot();
})