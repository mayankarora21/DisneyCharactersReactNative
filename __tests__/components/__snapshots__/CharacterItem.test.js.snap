// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <Text>
    CharacterName
  </Text>
  <Image
    source={
      {
        "uri": "https://static.wikia.nocookie.net/disney/images/8/80/Profile_-_627.png",
      }
    }
    style={
      {
        "borderRadius": 50,
        "height": 100,
        "resizeMode": "stretch",
        "width": 100,
      }
    }
  />
</View>
`;
